src/mixins/hasMarkers.js

```diff
@@ -0,0 +1,100 @@
+export const MARKERS_LIMIT = 3;
+export const MARKER_SIZE = 16;
+export const MARKER_PADDING = 4;
+
+/**
+ * Prepare nodes for the Markers to be inserted inside the shapes.
+ *
+ * @param string selector 
+ */
+export function markersMarkup(selector) {
+  let children = [];
+  for (let i = 0; i < MARKERS_LIMIT; i++) {
+    children.push({ tagName: 'image', selector: selector + '.' + i });
+  }
+  return {
+    tagName: 'g',
+    selector,
+    children,
+  };
+}
+
+/**
+ * Define attributes for the markers.
+ *
+ * @param string name 
+ * @param object base 
+ * @param int length 
+ */
+export function markersAttrs(name, base, inverted = 1) {
+  let attrs = {};
+  attrs[name] = base;
+  for (let i = 0; i < MARKERS_LIMIT; i++) {
+    attrs[name + '.' + i] = { 'ref-x': i * MARKER_SIZE * inverted, 'ref-y': 0, width: MARKER_SIZE, height: MARKER_SIZE };
+  }
+  return attrs;
+}
+
+export default {
+  data() {
+    return {
+      markers: {
+        topLeft: {},
+        topCenter: {},
+        topRight: {},
+        bottomLeft: {},
+        bottomCenter: {},
+        bottomRight: {},
+      },
+    };
+  },
+  watch: {
+    markers: {
+      inmediate: true,
+      deep: true,
+      handler() {
+        this.refreshMarkers();
+      },
+    },
+  },
+  methods: {
+    refreshMarkers(markers = this.markers) {
+      // Get shape size
+      const { width, height } = this.shape.size();
+      for (let position in markers) {
+        // Align markers (left-right)
```

Comment:
Instead of a comment like this, do you think you can break out the relevant code into a meaningful function? Something like `alignMarkersFromLeftToRight(position)`. You can apply the same technique to the other blocks in this function.